usagov/usagov-2021
# Usagov 1373 rework ssg post processing

## Jira Task

https://cm-jira.usa.gov/browse/USAGOV-1373

## Description

Modified the modifyHtml function in usagov_ssg_postprocessing/src/EventSubscriber/TomeEventSubscriber.php to use html5-php instead of DOMDocument(). This fixes the bug in dataLayer output for Spanish pages.

## Type of Changes

  - Add screenshot showing what it should look like

## Testing Instructions



### Requires New Config

### Requires New Content

### Validation Steps
- Empty your html directory
- Clear all caches
- Go to http://localhost/admin/config/tome/static/generate and submit the form
- After a couple of seconds of running this process, you should be able to locate this file: html/index.html. Open the file, search for 'href="/es' and verify that the url ends with a slash 
<img width="554" alt="image" src="https://github.com/usagov/usagov-2021/assets/39020809/c6f9a5ab-7161-49e5-aca5-6276fd22d80f">

- When this process is about to finish, you should have a folder titled "es" inside the html folder. Select any of those pages and inspect the index.html file (example: html/es/ayuda-estadounidenses-extranjero/index.html).
<img width="584" alt="image" src="https://github.com/usagov/usagov-2021/assets/39020809/9cb491da-5e9f-451b-bb57-1a768460b8e9">

- Search for the word "Taxonomy_Text_1" and verify that it says "Página principal", not "P&aacute;gina principal"
<img width="626" alt="image" src="https://github.com/usagov/usagov-2021/assets/39020809/0ff1e6d7-2b5f-4ba1-86f7-c6c03768a6b1">



## Security Review




## Reviewer Reminders
- Reviewed code changes
- Reviewed functionality
- Security review complete or not required

## Post PR Approval Instructions
Follow these steps as soon as you merge the new changes.

1. Go to the [USAGov Circle CI project](https://app.circleci.com/pipelines/github/usagov/usagov-2021).
2. Find the commit of this pull request.
3. Build and deploy the changes.
4. Update the Jira ticket by changing the ticket status to `Review in Test` and add a comment. State whether the change is already visible on [cms-dev.usa.gov](http://cms-dev.usa.gov/) and [beta-dev.usa.gov](http://beta-dev.usa.gov/), or if the deployment is still in process.
